File: backend/Models/Order.cs
namespace backend.Models;

public class Order : BaseModel
{
  public double TotalPrice { get; set; }
  public Guid UserId { get; set; }
  public User User { get; set; }
  public Guid? CouponId { get; set; }
  public Coupon Coupon { get; set; }
  public IEnumerable<OrderDetail> OrderDetails { get; set; }
  public Guid ContactId { get; set; }
  public Contact Contact { get; set; }
  public Guid CardId { get; set; }
  public Card Card { get; set; }
  public Guid AddressId { get; set; }
  public Address Address { get; set; }
}

File: admin/src/models/order.ts
import { Address, Card, Contact, Account } from "./account"
import { Product } from "./product"

export type Order = {
  id: string,
  totalPrice: number,
  userId: string;
  couponId?: string,
  orderDetails: OrderDetail[];
  contactId: string,
  contact: Contact,
  cardId: string,
  card: Card,
  addressId: string,
  address: Address,
  createAt: Date,
  user: Account
}

export type OrderDetail = {
  id: string,
  salePrice: string,
  quantity: number,
  orderId: string,
  productId: string,
  product: Product,
}

export type Coupon = {
  code: string;
  id: string,
  discount: number,
  startDate: Date,
  endDate: Date,
  totalUsed: number;
}